### PR TITLE
    Fix bugs and add info

### PR DESCRIPTION
**Pull request by zihengchang**
* Allow getting device parameters from ffmpeg pipeline
    * Add unit test with frame mode multi, supplementary info
    * Fix code conflict in hardware decoding mode

### CODE REVIEW

File: modules/gles/gl_stitcher.cpp [hunk 1]
```diff
@@ -348,6 +348,15 @@ StitcherImpl::export_dma_buffer (const SmartPtr<GLStitcher::StitcherParam> &para
     _dmabuf_handler[_camera_num]->set_opt_type (CopySSBO2Tex);
 
     XCamReturn ret = _dmabuf_handler[_camera_num]->execute_buffer (dma_param, false);
+#if DUMP_BUFFER
+    const char *file_name = "gl_stitcher_out_buf_7680_3840.nv12";
+    FILE* fbo_file = fopen (file_name, "wb");
+    if (fbo_file != NULL) {
+        fwrite (param->out_buf->map(), 7680 * 3840 * 3 / 2, 1, fbo_file);
```
Comment (zongwave) [inline modules/gles/gl_stitcher.cpp:355]: don't hardcode image resolution.

File: modules/gles/gl_stitcher.cpp [hunk 2]
```diff
@@ -1143,6 +1152,12 @@ GLStitcher::start_work (const SmartPtr<Parameters> &base)
     XCamReturn ret = XCAM_RETURN_NO_ERROR;
 
     SmartPtr<StitcherParam> param = base.dynamic_cast_ptr<StitcherParam> ();
+
+    ret = _impl->init_config (param);
```
Comment (zongwave) [inline modules/gles/gl_stitcher.cpp:1156]: can you guarantee init_config only be called once?

File: xcore/base/xcam_buffer.h [hunk 3]
```diff
@@ -103,6 +103,7 @@ struct _XCamVideoBuffer {
     XCamVideoBufferInfo   info;
     uint32_t              mem_type;
     int64_t               timestamp;
+    int                   hw_codec;
```
Comment (zongwave) [inline xcore/base/xcam_buffer.h:106]: can you use mem_type instead?
typedef enum {
    XCAM_MEM_TYPE_CPU,
    XCAM_MEM_TYPE_GPU,
    XCAM_MEM_TYPE_PRIVATE = 0x8000,
    XCAM_MEM_TYPE_PRIVATE_BO,
} XCamMemType;

**Review by zongwave — CHANGES_REQUESTED** [review]

File: modules/gles/gl_dma_buffer_handler.cpp [hunk 4]
```diff
@@ -23,6 +23,7 @@
 #include "gl_sync.h"
 
 #define INVALID_INDEX (uint32_t)(-1)
+#define DUMP_TEST 0
```
[the remark this answers is not in the document]
Comment (zihengchang) [reply modules/gles/gl_dma_buffer_handler.cpp:26]: Conditional compilation is more convenient if TEST_DEBUG is defined
https://github.com/intel/libxcam/blob/43d2aadcdc82da0a784c08f425f78bd37d6612f7/modules/gles/gl_dma_buffer_handler.cpp#L168

File: modules/gles/gl_stitcher.cpp [hunk 5]
```diff
@@ -1143,6 +1162,14 @@ GLStitcher::start_work (const SmartPtr<Parameters> &base)
     XCamReturn ret = XCAM_RETURN_NO_ERROR;
 
     SmartPtr<StitcherParam> param = base.dynamic_cast_ptr<StitcherParam> ();
+
+    if (!param->in_bufs[0].ptr ()) {
+        ret = _impl->init_dmabuf (param);
```
Comment (zongwave) [inline modules/gles/gl_stitcher.cpp:1167]: init_dmabuf_handler

File: tests/test-surround-view.cpp [hunk 6]
```diff
@@ -69,16 +70,16 @@ struct SVOutConfig {
...
-static void dump_dma_video_buf (SmartPtr<VideoBuffer> buf, const char *prefix_name, uint32_t idx)
+static void dump_dma_video_buf (SmartPtr<VideoBuffer> buf, const char *prefix_name)
 {
     char file_name[256];
     XCAM_ASSERT (prefix_name);
     XCAM_ASSERT (buf.ptr ());
 
     const VideoBufferInfo &info = buf->get_video_info ();
     snprintf (
-        file_name, 256, "%s-%dx%d.%05d.%s.yuv",
```
Comment (zongwave) [inline tests/test-surround-view.cpp:80]: to line 85: 
inside in function tex->dump_texture_image (file_name);
modify fopen (..., 'wb+')

**Review by zongwave — CHANGES_REQUESTED** [review]

**Comment by uartie** [comment]
stale, no activity for 7 months. closing.  feel free to reopen when issues resolved.

**Pull request closed without merging**